cmake/CMakeLists.txt

```diff
@@ -402,10 +402,26 @@ if (WIN32)
       endforeach()
     endif()
 else()
+  # Enabling this for Mac builds causes a sigabrt in several tests. Secondly the right option for Mac is -dead_strip
+  if (UNIX AND NOT APPLE)
+    foreach(type EXE SHARED)
+      set(CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO} --gc-sections")
```

Comment:
Please don't set them to all binaries. Because now we are having more and more third-party deps.
We may only set them on onnxruntime.so and the python binding so.